### PR TITLE
Delete `gtNewBlkOpNode` and `lvSimdBaseJitType`

### PR DESCRIPTION
**Pull request by SingleAccretion**
Both not needed anymore.

### CODE REVIEW

File: src/coreclr/jit/importer.cpp [hunk 1]
```diff
@@ -10577,7 +10581,7 @@ void Compiler::impImportBlockCode(BasicBlock* block)
 
                 op1 = gtNewLoadValueNode(layout, op1);
                 op2 = gtNewLoadValueNode(layout, op2);
-                op1 = gtNewBlkOpNode(op1, op2, ((prefixFlags & PREFIX_VOLATILE) != 0));
+                op1 = gtNewAssignNode(op1, op2);
```
Comment (SingleAccretion) [inline src/coreclr/jit/importer.cpp:10584]: Semantic change: `volatile.` is not allowed for `cpobj` per the specification, but we did accept it here.

We could accept both `volatile.` and `unaligned.` here with `impPrefixFlagsToIndirFlags` if there is a concern about this being a breaking change.

**Comment by SingleAccretion** [comment]
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=264762&view=ms.vss-build-web.run-extensions-tab) - one case on x86 where a SIMD local is no longer promoted.

@dotnet/jit-contrib

**Review by EgorBo — APPROVED** [review]

File: src/coreclr/jit/gentree.cpp [hunk 2]
```diff
@@ -8270,17 +8270,23 @@ GenTreeOp* Compiler::gtNewAssignNode(GenTree* dst, GenTree* src)
...
-
         SetOpLclRelatedToSIMDIntrinsic(dst);
         SetOpLclRelatedToSIMDIntrinsic(src);
     }
+#else  // TARGET_X86
+    // TODO-Cleanup: merge with the all-arch logic.
+    if (varTypeIsSIMD(src) && src->OperIs(GT_HWINTRINSIC, GT_CNS_VEC))
+    {
+        SetOpLclRelatedToSIMDIntrinsic(dst);
+    }
```
Comment (tannergooding) [inline src/coreclr/jit/gentree.cpp:8288]: What's the blocker for marking these for `src` on x86?

File: src/coreclr/jit/gschecks.cpp [hunk 3]
```diff
@@ -411,20 +411,9 @@ void Compiler::gsParamsToShadows()
...
-
-#ifdef FEATURE_SIMD
-        shadowVarDsc->lvUsedInSIMDIntrinsic = varDsc->lvUsedInSIMDIntrinsic;
-        if (varTypeIsSIMD(varDsc))
-        {
-            CorInfoType simdBaseJitType = varDsc->GetSimdBaseJitType();
-            shadowVarDsc->SetSimdBaseJitType(simdBaseJitType);
-        }
-#endif
+        var_types type            = varTypeIsSmall(varDsc->TypeGet()) ? TYP_INT : varDsc->TypeGet();
```
Comment (tannergooding) [inline src/coreclr/jit/gschecks.cpp:414]: Can this not be `genActualType(varDsc)`?

File: src/coreclr/jit/importer.cpp [hunk 4]
```diff
@@ -8751,17 +8750,10 @@ void Compiler::impImportBlockCode(BasicBlock* block)
...
-                            {
-                                newObjInit = gtNewBlkOpNode(newObjDst, gtNewIconNode(0));
-                            }
-                            else
-                            {
-                                newObjInit = gtNewAssignNode(newObjDst, gtNewZeroConNode(lclDsc->TypeGet()));
-                            }
-                            impAppendTree(newObjInit, CHECK_SPILL_NONE, impCurStmtDI);
+                            GenTree* newObjInit =
+                                gtNewZeroConNode((lclDsc->TypeGet() == TYP_STRUCT) ? TYP_INT : lclDsc->TypeGet());
```
Comment (tannergooding) [inline src/coreclr/jit/importer.cpp:8754]: Same here, can this not use `genActualType`?

**Review by tannergooding — APPROVED** [review]

File: src/coreclr/jit/importer.cpp (hunk 4 above)
Comment (SingleAccretion) [reply src/coreclr/jit/importer.cpp:8754]: No need - `gtNewZeroConNode` will normalize inside itself. And of course for `TYP_STRUCT`, it wouldn't work.

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**